### DISCUSSION TITLE
MLflow recipes and transformers that return sparse matrix

### DISCUSSION THREAD

**Discussion by hluu480** [mlflow/mlflow#7687] [Q&A]
I'm new to MLflow Recipes (new to MLflow in general) but from what I have experimented with so far, the transform step will fail for sklearn transformers that return a sparse matrix such as HashingVectorizer or OneHotEncoder with sparse=True. Digging into the code a bit, I see that the transformed features are ultimately persisted to a local parquet file, but alas pyarrow currently doesn't support sparse data structures. Has anyone gotten MLflow Recipes to work with sparse data?

**Comment by JHibbard** [comment]
Hi hluu480,

Would you be able to share you MLflow version and a minimal code example? This may be a bug/limitation and it'd be great to track this in the MLflow's issues. Thank you for sharing your roadblock, hopefully we can get this unstuck!

Jim
